sdks/java/io/kafka/upgrade/src/main/java/org/apache/beam/sdk/io/kafka/upgrade/KafkaIOTranslation.java

```diff
@@ -137,7 +137,7 @@ public Row toConfigRow(Read<?, ?> transform) {
         fieldValues.put("topics", transform.getTopics());
       }
 
-      if (transform.getTopicPartitions() != null) {
+      if (transform.getTopicPartitions() != null && !transform.getTopicPartitions().isEmpty()) {
```

Comment:
IIRC there's a nullOrEmpty utility function.

Reply:
I couldn't find one or add one that also satisfies that checker-framework nullness check. So leaving as is.